### PR TITLE
Update to Swift 3.0

### PR DESCRIPTION
**Pull request by PtitChevreuil**
I'm currently starting a project with the VIPER design pattern. I found this ToDoList example really useful. For this reason I updated it to Swift 3.0.

All tests of the NSCalendar extension are still passing

### CODE REVIEW

**Comment by JeffGilbert-MM** [comment]
Thank you for taking the time to do this.

**Comment by ronlisle** [comment]
I found 1 line that hadn't been updated, and was commented out, and have added a comment for it.
(NSCalendar+CalendarAdditions.swift Line 50)
Other than that, it builds, tests, and runs ok. Thanks for doing that for us!

File: VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift [hunk 1]
```diff
@@ -8,129 +8,130 @@
 
 import Foundation
 
-extension NSCalendar {
-    class func gregorianCalendar() -> NSCalendar {
-        return NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
+extension Calendar {
+    static func gregorianCalendar() -> Calendar {
```
Comment (PtitChevreuil) [inline VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift]: I realize I didn't follow new naming conventions at all ! "gregorian" is definitely a better name

File: VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift [hunk 2]
```diff
@@ -8,129 +8,130 @@
...
-        let newComponents = components(([.Year, .Month, .Day]), fromDate: date)
+    func yearMonthDayComponentsFromDate(_ date: Date) -> DateComponents {
+        let newComponents = dateComponents(([.year, .month, .day]), from: date)
         return newComponents
     }
     
-    func compareInteger(left: Int, right: Int) -> NSComparisonResult {
-        var result = NSComparisonResult.OrderedDescending
+    func compareInteger(_ left: Int, right: Int) -> ComparisonResult {
+        var result = ComparisonResult.orderedDescending
```
Comment (PtitChevreuil) [inline VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift]: It is possible. But we have to tell the compilator that result is a `ComparisonResult` variable:

> var result: ComparisonResult = .orderedDescending

File: VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift [hunk 3]
```diff
@@ -8,129 +8,130 @@
...
-    class func gregorianCalendar() -> NSCalendar {
-        return NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
+extension Calendar {
+    static func gregorianCalendar() -> Calendar {
+        return Calendar(identifier: Calendar.Identifier.gregorian)
     }
     
-    func dateWithYear(year: Int, month: Int, day: Int) -> NSDate {
-        let components = NSDateComponents()
+    func dateWithYear(_ year: Int, month: Int, day: Int) -> Date {
```
Comment (PtitChevreuil) [inline VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift]: > dateWith(year: Int, ...)

 ? ^^

File: VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift (hunk 2 above)
Comment (JeffGilbert-MM) [inline VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift]: You can leave it like it is.

**Comment by tariq235** [comment]
@JeffGilbert-MM Can we merge this branch if we are done with swift 3 changes ?

**Comment by JeffGilbert-MM** [comment]
It has been merged.

**Pull request merged**